packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -307,10 +307,10 @@ export const DEFAULT_GRID_OPTIONS: GridOptions = {
   sortingMode: FeatureModeConstant.client,
   sortingOrder: ['asc', 'desc', null],
   columnTypes: DEFAULT_COLUMN_TYPES,
-  disableColumnMenu: process.env.NODE_ENV === 'production',
-  disableColumnFilter: process.env.NODE_ENV === 'production',
-  disableColumnSelector: process.env.NODE_ENV === 'production',
-  hideToolbar: process.env.NODE_ENV === 'production',
+  disableColumnMenu: process.env.ENABLE_EXPERIMENTAL !== 'true',
+  disableColumnFilter: process.env.ENABLE_EXPERIMENTAL !== 'true',
+  disableColumnSelector: process.env.ENABLE_EXPERIMENTAL !== 'true',
+  hideToolbar: process.env.ENABLE_EXPERIMENTAL !== 'true',
```

Comment:
How about we ALSO allow to enable_experimantal by using chrome debugger?

Reply:
Haha, one step at a time? :) What use case to you have in mind?

Reply:
I think it would be great because we need to allow feature that are experimental to be toggled on and off in an easy way.

incremental development and smaller PR would be much easier to achieve with this toggle. 

For example, I added the drag icon on the column picker but not implemented the drag yet, so I'm thinking of removing it but if we have it under the experimental feature then we can leave it.

Reply:
> For example, I added the drag icon on the column picker but not implemented the drag yet, so I'm thinking of removing it but if we have it under the experimental feature then we can leave it.

What prevents you from using `process.env.ENABLE_EXPERIMENTAL` in this context?

One thing I'm not so sure about if we should use `process.env.CONTEXT === 'production'` or `process.env.PULL_REQUEST === 'false'` from the Netlify env. Basically, should we see experimental features when looking at https://master--material-ui-x.netlify.app/components/data-grid (master)?

Reply:
that's why I want to create an abstraction so we hide where it gets the value from, and it can get it from different sources. Ideally, experimental features should be easily turned on in any environment.

Reply:
To be cautious with the bundle size cost and the extra complexity. I think that we should design the system to have the least amount of flexibility possible.

Reply:
Ideally you want to tree shake experimental code of the production bundle

Reply:
I believe that because we use `process.env.ENABLE_EXPERIMENTAL`, the code wrapped won't be included in the developers' bundles: https://codesandbox.io/s/material-demo-forked-991f4?file=/package.json

Reply:
Can we put it in a global var? ie
```
window._mui_experimental = process.env.ENABLE_EXPERIMENTAL;
```
Then to turn on and off, the experimental mode we can use chrome debugger

Reply:
I don't want to put `process.env.ENABLE_EXPERIMENTAL` in components, it would be horrible.
We need a nice wrapper, maybe something like `Feature.experimental && <Component />`

Reply:
Maybe it would work. I still understand why we would want to do it. There are already props do enable the feature. You can open the demos in codesandbox. 

I have opened this pull request to solve this concret case:

- a pull request was opened to change an experimental feature
- I did have time to checkout the branch locally.

This could also be solved with this alternative path: add a documentation for it, under an experiment label. Actually, I prefer the last option.

Reply:
> it would be horrible

Why?